### PR TITLE
Made non blocking workflows, TSValue doc.

### PR DESCRIPTION
**Pull request by Cooksen**
Haven't done mock tests yet.

### CODE REVIEW

File: decanter_ai_sdk/decanter_client.py [hunk 1]
```diff
@@ -0,0 +1,623 @@
+from io import StringIO
```
Comment (jwolf9) [inline decanter_ai_sdk/decanter_client.py]: I think the name can still be client. So we have `client` and `non_blocking_client`

File: decanter_ai_sdk/experiment.py [hunk 2]
```diff
@@ -168,3 +168,8 @@ def experiment_info(self) -> Dict:
             "created_at": self.created_at,
             "completed_at": self.completed_at,
         }
+
+
+class ExperimentResult(BaseModel):
+    result: Experiment
+    status: str
```
Comment (jwolf9) [inline decanter_ai_sdk/experiment.py:175]: would be easier to use if it's an enum

File: decanter_ai_sdk/non_blocking_client.py [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+from .client import AbstractClient
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Client(AbstractClient):
+    """
+    Handle client side actions.
```
Comment (jwolf9) [inline decanter_ai_sdk/non_blocking_client.py:25]: explain a little bit how it's different from the normal client

File: decanter_ai_sdk/non_blocking_client.py [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+            holdout_percentage (int)
+                Holdout percentage for experiment.
+
+        Returns:
+        ----------
+            (`~decanter_ai_sdk.web_api.experiment.Experiment`)
+                Experiment id.
+        """
+
+        data_column_info = self.api.get_table_info(table_id=experiment_table_id)
```
Comment (jwolf9) [inline decanter_ai_sdk/non_blocking_client.py:197]: you want to factor out all this duplicate code to helper functions that can be shared between the two clients. There should never be this much duplicate code. It makes it more difficult when it needs to be modified in the future

File: decanter_ai_sdk/prediction.py [hunk 5]
```diff
@@ -23,3 +23,8 @@ def get_predict_df(self) -> pd.DataFrame:
 
         """
         return self.predict_df
+
+
+class PredictionResult(BaseModel):
+    result: Prediction
```
Comment (jwolf9) [inline decanter_ai_sdk/prediction.py:29]: should be optional, because it will be None if it is not finished yet

Comment (jwolf9) [reply decanter_ai_sdk/prediction.py:29]: make the same change for experiment

File: examples/non_blocking_iid_example.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
+from time import sleep
+from decanter_ai_sdk.non_blocking_client import Client
```
Comment (jwolf9) [inline examples/non_blocking_iid_example.py:2]: The name of the client should be different for blocking and non blocking. I think `Client` and `NonBlockingClient` is ok

File: examples/non_blocking_iid_example.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        evaluator=ClassificationMetric.AUC,
+        custom_feature_types={
+            "Pclass": DataType.categorical,
+            "Parch": DataType.categorical,
+        },
+        max_model=5,
+        algos=["DRF", "GBM", IIDAlgorithms.DRF],
+    )
+
+    # print("This will show info of the experiment:\n", experiment.experiment_info())
```
Comment (jwolf9) [inline examples/non_blocking_iid_example.py:66]: delete or uncomment

File: examples/non_blocking_iid_example.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        "This will show the id of tested ../data:",
+        prediction.result.attributes["table_id"],
+        "\n",
+    )
+    print(
+        "Head of the prediction ../data:\n", prediction.result.get_predict_df().head()
+    )
+
+
+test_iid()
```
Comment (jwolf9) [inline examples/non_blocking_iid_example.py:110]: put this in a main function

Comment (jwolf9) [reply examples/non_blocking_iid_example.py:110]: same for ts

File: examples/non_blocking_ts_example.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
+from time import sleep
+from decanter_ai_sdk.non_blocking_client import Client
+import os
+from decanter_ai_sdk.enums.data_types import DataType
+from decanter_ai_sdk.enums.time_units import TimeUnit
+from decanter_ai_sdk.enums.evaluators import RegressionMetric
+from decanter_ai_sdk.enums.algorithms import TSAlgorithms
+
+
+def test_iid():
```
Comment (jwolf9) [inline examples/non_blocking_ts_example.py:10]: `test_ts()`

File: examples/non_blocking_ts_example.py [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        "This will show the id of tested ../data:",
+        prediction.result.attributes["table_id"],
+        "\n",
+    )
+    print(
+        "Head of the prediction ../data:\n", prediction.result.get_predict_df().head()
+    )
+
+
+test_iid()
```
Comment (jwolf9) [inline examples/non_blocking_ts_example.py:110]: `test_ts()`

**Review by jwolf9 — CHANGES_REQUESTED** [review]

File: decanter_ai_sdk/non_blocking_client.py (hunk 4 above)
Comment (Cooksen) [reply decanter_ai_sdk/non_blocking_client.py:197]: Roger that!

**Review by jwolf9 — APPROVED** [review]

**Pull request merged**